.pre-commit-config.yaml

```diff
@@ -74,9 +74,10 @@ repos:
   - id: mypy
     exclude: ^(doc/|tests/|examples/|pyvista/ext/|examples_trame/)
     additional_dependencies: [
-    "mypy-extensions==1.0.0",
-    "toml==0.10.2",
-    "numpy",
+      "mypy-extensions==1.0.0",
+      "toml==0.10.2",
+      "numpy>=1.21",
+      "https://github.com/pyvista/npt-promote/tarball/main"
```

Comment:
`npt-promote` is a new pyvista repo. It has not yet been added to PyPI. If added, this line could be changed to "npt-promote".

Reply:
If possible, I would like to merge this after releasing version 0.1. Is it OK to register it with PyPI?

Reply:
Yes please register it if you can. I wasn't able to a few weeks ago since there was a block on making new PyPI accounts.

Reply:
Looks like registration for PyPI is open again, I'll make some minor changes to `pyvista/npt-promote` register it, and push a new commit here with a fixed version of the plugin.

Reply:
I have updated `npt-promote` with CI testing. I think's ready for a first publishing (e.g. as version 0.1). I went to publish on `PyPI` but it seems fairly involved, requiring several builds to be submitted for each platform and python version  (I have no prior experience with publishing to PyPI). How is this process typically done within `pyvista` ?

Also, I'm not sure if the publishing account matters? Is it better to publish it from the same account from which `pyvista` is also published?

Reply:
Below is where we upload to PyPI. It is triggered when released with the tag. Let's start by releasing using Python 3.9 version.

https://github.com/pyvista/pyvista/blob/f37b5da5c9de3bff2e69abebc147155eab8e25db/.github/workflows/testing-and-deployment.yml#L221-L228

Since `npt-promote` is your project, you can publish it under your account :)

Reply:
Thanks, that was helpful. `npt-promote==0.1` has been published.